crates/noirc_evaluator/src/ssa/node.rs

```diff
@@ -260,19 +279,20 @@ impl std::fmt::Display for Instruction {
 #[derive(Debug, Clone, Copy)]
 pub enum NodeEval {
     Const(FieldElement, ObjectType),
-    Idx(arena::Index),
+    Instruction(NodeId),
```

Comment:
Why did you rename Idx into Instruction?
AFAIK, the underlying NodeId is not necessarily an instruction, it can also be a Variable.

Reply:
Ah, I'll change it to something more explicit like `VarOrInstruction` in that case